portal/lib/download_template.php

```diff
@@ -165,6 +165,22 @@ function doSubs($s)
         } elseif (keySearch($s, '{SignaturesRequired}')) {
             $sigfld = '<script>page.signaturesRequired=true;var signMsg=' . xlj("A signature is required for this document. Please sign document where required") . ';</script>' . "\n";
             $s = keyReplace($s, $sigfld);
+        } elseif (preg_match('/^{(Questionnaire):(.*)}/', substr($s, $keyLocation), $matches)) {
+            $q_id = $matches[2];
+            $keyLength = strlen($matches[0]);
+            $src = './../questionnaire_template.php?isPortal=1&qId=' . urlencode($q_id);
+            $sigfld = "<script>page.isQuestionnaire=1;page.questionnaireName=" . js_escape($q_id) . "</script>";
+            $sigfld .= "<iframe id='questionnaire' class='questionnaires' style='height:100vh;width:100%;border:0;' src='" . $src . "'></iframe>";
```

Comment:
Is $src escaped further up in this file?

Reply:
no need imo

Reply:
should have attr() around $src (otherwise could possibly sneak cross scripting via $q_id)